pkg/endpoint/identifiers.go

```diff
@@ -46,34 +43,29 @@ func (e *Endpoint) HumanStringLocked() string {
 // GetK8sNamespaceAndPodName returns the corresponding namespace and pod
 // name for this endpoint.
 func (e *Endpoint) GetK8sNamespaceAndPodName() string {
-	e.unconditionalRLock()
-	defer e.runlock()
-
 	return e.getK8sNamespaceAndPodName()
```

Comment:
Do we still need both exported and un-exported function of the "same" name and function?

Reply:
hah. No.